### PR TITLE
[Rados] Cluster fill method for Rados Performance workflow

### PR DESCRIPTION
**Pull request by harshkumarRH**
Building blocks for Performance evaluation workflow at RADOS layer using radosbench tool

This PR adds the module to warm up the cluster to an initial desired percentage.

Signed-off-by: Harsh Kumar <hakumar@redhat.com>

### CODE REVIEW

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -3849,3 +3849,16 @@ def get_osd_details(self, osd_id: int):
         details = out["nodes"][0]
         log.debug(f" OSD : {osd_id} details are : {details}")
         return details
+
+    def get_osd_size(self, osd_id):
+        """
+        Method returns the size of the OSD
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:3855]: maybe just elaborate in the docstring that the size is returned in KB

File: ceph/rados/perf_workflow.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        Returns:
+            obj count to be written by the client
+        """
+        # total available space for the pool
+        pool_stat = self.rados_obj.get_cephdf_stats(pool_name=pool_name)
+        max_avail_kb = pool_stat["stats"]["max_avail"]
+        # max_avail_mb = floor(max_avail_kb / 1 << 10)
+
+        # amount of data to be written to the pool
+        total_write_data = floor(max_avail_kb * fill_percent / 100)
```
Comment (pdhiran) [inline ceph/rados/perf_workflow.py:78]: Are we taking into consideration the replica size? for calculating total amount of data to be written?

Comment (pdhiran) [reply ceph/rados/perf_workflow.py:78]: If for now, we are only targeting RE pools, we can add that in a doc string. It's good enough to start from RE pools

File: ceph/rados/perf_workflow.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        obj_size_kb_int = int(obj_size[:-1])
+        if "M" in obj_size:
+            obj_size_kb_int *= 1 << 10
+        obj_count = floor(total_write_client / obj_size_kb_int)
+
+        pool_multi = self.get_pool_multiplier(pool_name=pool_name)
+        obj_count = int(obj_count / pool_multi)
+        return obj_count
+
+    def get_pool_multiplier(self, pool_name) -> 3:
```
Comment (pdhiran) [inline ceph/rados/perf_workflow.py:93]: the default return type should be int?? `get_pool_multiplier(self, pool_name) -> int:`

File: ceph/rados/perf_workflow.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        Method to calculate the object count multiplier depending
+        on the pool type, size and min size
+        Args:
+            pool_name:
+        Returns:
+            multiplier value (int)
+        """
+        pool_detail = self.rados_obj.get_pool_details(pool=pool_name)
+        if pool_detail["erasure_code_profile"]:
+            return pool_detail["size"] / pool_detail["min_size"]
```
Comment (pdhiran) [inline ceph/rados/perf_workflow.py:104]: Maybe we should round the number up to max 2 decimal points.

File: ceph/rados/perf_workflow.py (hunk 2 above)
Comment (harshkumarRH) [reply ceph/rados/perf_workflow.py:78]: The pool multiplier or maybe I should rename it to pool divider has been written to take care of replica size and facilitate object count calculation accordingly

File: ceph/rados/perf_workflow.py (hunk 4 above)
Comment (harshkumarRH) [reply ceph/rados/perf_workflow.py:104]: Any reason that would be necessary? The more the decimal points, more the precision

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (harshkumarRH) [reply ceph/rados/core_workflows.py]: Added

File: ceph/rados/perf_workflow.py (hunk 3 above)
Comment (harshkumarRH) [reply ceph/rados/perf_workflow.py]: Changed the return type to float

**Review by neha-gangadhar — COMMENTED** [review]
lgtm

**Review by pdhiran — APPROVED** [review]
Looks good.

**Pull request merged**